assets/app/view/game/issue_shares.rb

```diff
@@ -10,18 +10,18 @@ class IssueShares < Snabberb::Component
       def render
         @step = @game.round.active_step
         @entity = @game.current_entity
-
+        @current_actions = @game.round.actions_for(@entity)
         children = []
 
-        if @step.current_actions.include?('sell_shares')
-          children << render_shares('Issue', @step.issuable_shares(@entity), Engine::Action::SellShares)
+        if @current_actions.include?('sell_shares')
+          children << render_shares('Issue', @game.issuable_shares(@entity), Engine::Action::SellShares)
```

Comment:
why does this need to be in game, can't you just do this directly here?

round.step_for(entity, 'sell_shares').issuable_shares

Reply:
When we discussed in slack, you mentioned the client code shouldn't have to know much detail about the server code. In this case, should the client have to know that issuable_shares is step dependent? I leaned towards no and abstracted via the game class; however, both approaches are fine by me. Let me know which you prefer.

Reply:
sounds fine we'll keep